cylc/flow/scripts/release.py

```diff
@@ -54,23 +64,41 @@ def get_option_parser():
     parser = COP(
         __doc__, comms=True, multitask=True,
         argdoc=[
-            ("REG", 'Suite name'),
-            ('[TASK_GLOB ...]', 'Task matching patterns')])
+            ('REG', "Workflow name"),
+            ('[TASK_GLOB ...]', "Task matching patterns")]
+    )
+
+    parser.add_option(
+        "--all",
+        help=(
+            "Release all held tasks and remove the 'hold after cycle point', "
+            "if set."),
+        action="store_true", dest="release_all")
 
     return parser
 
 
 @cli_function(get_option_parser)
-def main(parser, options, suite, *task_globs):
-    suite = os.path.normpath(suite)
-    pclient = SuiteRuntimeClient(suite, timeout=options.comms_timeout)
+def main(parser, options, workflow, *task_globs):
+
+    if options.release_all:
+        if task_globs:
+            raise UserInputError("Cannot combine --all with TASK_GLOB(s).")
+    else:
+        if not task_globs:
+            raise UserInputError(
+                "Missing arguments: TASK_GLOB [...]. "
+                "See `cylc release --help`.")
+
+    workflow = os.path.normpath(workflow)
+    pclient = SuiteRuntimeClient(workflow, timeout=options.comms_timeout)
 
     mutation_kwargs = {
         'request_string': MUTATION,
         'variables': {
-            'wFlows': [suite],
+            'wFlows': [workflow],
             'tasks': list(task_globs),
-        }
+        }  # If --all, the list of task_globs is empty
```

Comment:
> So does this mean that if you call the `release` mutation with an empty list of task globs it will release everything?

Yes

> I think `cylc release --all` is equivalent to `cylc release '*'` so could just stick with the glob expansion for this use case.

Not quite - `--all` will unset the hold point (if it was set), whereas `'*'` will not